File: examples/python/stateful-counter/Counter.runar.py
from runar import StatefulSmartContract, Bigint, public, assert_

class Counter(StatefulSmartContract):
    count: Bigint

    def __init__(self, count: Bigint):
        super().__init__(count)
        self.count = count

    @public
    def increment(self):
        self.count += 1

    @public
    def decrement(self):
        assert_(self.count > 0)
        self.count -= 1

File: examples/ruby/stateful-counter/Counter.runar.rb
require 'runar'

class Counter < Runar::StatefulSmartContract
  prop :count, Bigint

  def initialize(count)
    super(count)
    @count = count
  end

  runar_public
  def increment
    @count += 1
  end

  runar_public
  def decrement
    assert @count > 0
    @count -= 1
  end
end
